build-system/pr-check/safari-tests.js

```diff
@@ -0,0 +1,84 @@
+/**
```

Comment:
Personally I still prefer splitting each test type to its own shard, even if it costs a little extra. It would just make things easier to scan quickly (`Safari Unit Tests - PASSED; Safari End-to-End Tests - FAILED` tells me immediately which tests passed/failed, while `Safari Tests - FAILED` - does not). Not feeling too strongly about this though

Reply:
I considered this, but decided to use the same VM due to the timing.

- It takes ~2 full minutes just to set up a macOS VM (they're not nearly as fast as docker).
- After that, unit / integration / e2e tests each take less than 1 minute to run.
- Splitting them across three jobs will nearly double the CPU utilization on macOS VMs.
- macOS VMs are 5x as expensive as the equivalent docker VM (50 credits / min vs. 10 credits / min). 

That said, I agree it's worth checking if other inexpensive optimizations are possible. Lemme revisit this in a follow up PR.